### ISSUE TITLE
major spam attack by a bot

### ISSUE THREAD

**Issue by tfoote** [ros-infrastructure/answers.ros.org#76]
They're creating new users and posting one question per user. 

It also seems that the questions for new users can link and it doesn't have the nofollow attribute.

**Comment by tfoote** [comment]
We've turned off password based registration for the moment.

**Comment by trainman419** [comment]
I'm seeing this again today

**Comment by tfoote** [comment]
There were some 50-100 spam posts again. From 4-5 users.

**Comment by evgenyfadeev** [comment]
Hi Tully,

I'm thinking what would be the most effective method.
Perhaps user moderation? For example a user with a "watched" status
 is allowed to make one post after which we'd have to decide whether to
allow posting anything else
or not.

There is also Captcha, but supposedly the the ReCaptcha has been solved and
can
be overcome with automation.

Other possibility - limit account creation per IP address. It may be too
restrictive though.

The issue is that no matter what, the users will be able to post something
manually. So I think we'd have to go the "moderation" route.

Please let me know what you think.

Thank you,
Evgeny.

On Tue, May 27, 2014 at 1:21 AM, Tully Foote notifications@github.comwrote:

> There were some 50-100 spam posts again. From 4-5 users.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/ros-infrastructure/answers.ros.org/issues/76#issuecomment-44234668
> .

## 

Askbot
Valparaiso, Chile
skype: evgeny-fadeev

**Comment by evgenyfadeev** [comment]
I will give captcha a spin, but it's possible we'll need a set of features
to reduce spam.

On Tue, May 27, 2014 at 1:39 AM, Evgeny Fadeev evgeny.fadeev@gmail.comwrote:

> Hi Tully,
> 
> I'm thinking what would be the most effective method.
> Perhaps user moderation? For example a user with a "watched" status
>  is allowed to make one post after which we'd have to decide whether to
> allow posting anything else
> or not.
> 
> There is also Captcha, but supposedly the the ReCaptcha has been solved
> and can
> be overcome with automation.
> 
> Other possibility - limit account creation per IP address. It may be too
> restrictive though.
> 
> The issue is that no matter what, the users will be able to post something
> manually. So I think we'd have to go the "moderation" route.
> 
> Please let me know what you think.
> 
> Thank you,
> Evgeny.
> 
> On Tue, May 27, 2014 at 1:21 AM, Tully Foote notifications@github.comwrote:
> 
> > There were some 50-100 spam posts again. From 4-5 users.
> > 
> > —
> > Reply to this email directly or view it on GitHubhttps://github.com/ros-infrastructure/answers.ros.org/issues/76#issuecomment-44234668
> > .
> 
> ## 
> 
> Askbot
> Valparaiso, Chile
> skype: evgeny-fadeev

## 

Askbot
Valparaiso, Chile
skype: evgeny-fadeev

**Comment by tfoote** [comment]
​Lets give it a try. If we can make it just a little harder for the
spammers they might go elsewhere.

**Comment by skohlbr** [comment]
It appears the spam attacks in recent days are all in arabic and targetting "customers" in egypt. Would it make sense to install filters that prevent posting of messages using majorly non-english alphabets?

**Comment by BennyRe** [comment]
According to [this changelog](http://askbot.org/doc/changelog.html) askbot supports [Akismet](http://akismet.com/).

Could this help to prevent the spam?

**Comment by dlaz** [comment]
Is there any sort of rate limiting for posting questions? Limiting single users to only posting a few questions per hour could at least make moderation a little more manageable for the community.

**Comment by evgenyfadeev** [comment]
There isn't, but actually rate limiting is not a problem to bypass, you can
just schedule spam posts by script.
Rate limiting can help in some cases.

I'm almost done with recaptcha and implementing "watched" user status,
where "watched" users
will be forced to use recaptcha on every post (also on registration).

On Wed, May 28, 2014 at 12:53 PM, Dan Lazewatsky
notifications@github.comwrote:

> Is there any sort of rate limiting for posting questions? Limiting single
> users to only posting a few questions per hour could at least make
> moderation a little more manageable for the community.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/ros-infrastructure/answers.ros.org/issues/76#issuecomment-44426305
> .

## 

Askbot
Valparaiso, Chile
skype: evgeny-fadeev

**Comment by tfoote** [comment]
Recaptcha for posting sounds good. And "watched" means that they're below X karma? The spammers seem to come back almost every night and can get through the open ID registration requirement.

**Comment by evgenyfadeev** [comment]
Watched atm means - uses captcha, also there will be a threshold
for auto-approving user (change status from "watched" to "approved")
when their reputation crosses the margin from below.

So it will be possible to force users to use captcha.

On Wed, May 28, 2014 at 2:47 PM, Tully Foote notifications@github.comwrote:

> Recaptcha for posting sounds good. And "watched" means that they're below
> X karma? The spammers seem to come back almost every night and can get
> through the open ID registration requirement.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/ros-infrastructure/answers.ros.org/issues/76#issuecomment-44441124
> .

## 

Askbot
Valparaiso, Chile
skype: evgeny-fadeev

**Comment by evgenyfadeev** [comment]
Enabled captcha and made all users with karma > 10 bypass captcha.
Users with "watched" status and anonymous visitors will have to pass the captcha test:
- when registering
- asking, answering, editing question or answer
- when sending feedback

Users who reach karma 10 (adjustable) will become auto-approved.

If you want to force users to use captcha - change their status to "watched".

Let's see how this works.

**Comment by evgenyfadeev** [comment]
sorry, disabled for now, will update shortly.

**Comment by evgenyfadeev** [comment]
Should work now. We should now see questions and answers posted by new and low rep users (<10). Users should also be able to register and edit posts.

It would be best to test this on a live site, maybe someone could try creating a new account and make some reasonable questions/answers and edit them?

Works on my development computer. On answers.ros.org I've tested feedback form and registration.

**Comment by tfoote** [comment]
We're still getting 2-3 attacks per day. And they're posting 3 to 20 posts each. 

I tried out using the captcha, and it's a little awkward when you fail it that it reloads the page and there's no indication to retry, and doing it from an answer section the answer is folded up with your content missing. 

And I did not seem to be able to register without an OpenID yet. I guess that's not enabled yet? It's hard to test the registration by creating more openID accounts.

**Comment by tfoote** [comment]
I did create a test account, and was able to post a question. I was not able to post a question with an external link. (However it did not give any feedback as to why it would not post. Whereas if I failed the captcha it told me to try again in red.)

**Comment by tfoote** [comment]
It did require the captcha for registering and posting. I think we may be up against a real person.  I'm not sure what else we can do. Though the useability comments above would be nice to fix.

**Comment by evgenyfadeev** [comment]
The immediate plan is to implement pre-moderation of new users posts,
tools for mass clean-up and the IP blocking.

Best regards,
Evgeny.

On Tue, Jun 10, 2014 at 4:17 PM, Tully Foote notifications@github.com
wrote:

> It did require the captcha for registering and posting. I think we may be
> up against a real person. I'm not sure what else we can do. Though the
> useability comments above would be nice to fix.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/ros-infrastructure/answers.ros.org/issues/76#issuecomment-45658894
> .

## 

Askbot
Valparaiso, Chile
skype: evgeny-fadeev

**Comment by tfoote** [comment]
I'd strongly prefer to make the cleanup easier. Moderating all new users
will greatly decrease the initial impression on the users and add more
overhead and another stream to require everyone to monitor.

If we could make it such that it was just a few clicks to ban a user and
delete all their posts that would be fine. Right now when I get 30 incoming
posts. I need to click on each one to clear it from the review queue, and I
also have to check that the user has been blocked. And then check that all
their posts have been cleared. Adding this sort of thing to the drop down
on the review page would be great.

If the review queue could make the user status and post status more visible
such that if another admin has already deleted and blocked the user I could
know that and just mark as read that would be very helpful.

**Comment by gavanderhoorn** [comment]
@tfoote wrote:

>  I think we may be up against a real person. I'm not sure what else we can do.

If we are they are persistent: just removed 20+ posts by three different accounts (_yasooo2014_ and _saaraaa_. Forgot the 3rd one).

**Comment by skohlbr** [comment]
What we´ve seen in recent weeks was mainly arabic spam, so just mentioning again my idea of preventing posting of (majorly) non-english alphabet postings. Not sure about technical feasibility, but should prevent the current style of spam attacks (or force them to switch to english ;) ).

**Comment by evgenyfadeev** [comment]
Deployed feature - "block user and delete all content" in "profile->moderation". Please try this.

**Comment by tfoote** [comment]
Great! It's there and seems to be working. 

I think the last useability issue is that moderation/flag review area doesn't work well with multiple users. If I go to review my queue and someone else has already cleaned up I have no way to know that the flag has been dealt with by another moderator. 

What I'd love to see would be a single queue with individual new vs viewed, and a history next to each item for "flagged by X (+ Y + Z)( deleted by moderator W | approved by moderator V) 

And preferably the moderation result could also be overturned from that view keeping the history of actions on the post. One of the issues I've had is if a post is accidentally moderated/deleted. It dissapears from the review queue, and then you need to remember the user profile to undelete the question or answer. (Since you can find it on their user profile still)

**Comment by tfoote** [comment]
Ohh, it would also be great to see the user Moderation Status, aka "User Blocked" on the review page too.

**Comment by tfoote** [comment]
Just randomly I spot checked one of the spam numbers and it appears to have gotten onto google. 
![spam_on_google](https://cloud.githubusercontent.com/assets/447804/3350679/923eef64-f9c9-11e3-8d6a-6f39bb875fbf.png) The [link google points to](http://answers.ros.org/question/173167/lw-yz-trf-rqm-twkyl-bwsh-blskndry01227713067tsl-ln/) is invalid now when I browse, not logged in as an admin. However are we sure Google's not finding it still? 

It appears that whoever is doing this is doing it all over the internet from the search too. There's posts on youtube, vimeo, adobe forums, facebook...

**Comment by evgenyfadeev** [comment]
Maybe we automatically add blocked content into the robots.txt?

On Sun, Jun 22, 2014 at 2:09 AM, Tully Foote notifications@github.com
wrote:

> Just randomly I spot checked one of the spam numbers and it appears to
> have gotten onto google.
> [image: spam_on_google]
> https://cloud.githubusercontent.com/assets/447804/3350679/923eef64-f9c9-11e3-8d6a-6f39bb875fbf.png
> The link google points to
> http://answers.ros.org/question/173167/lw-yz-trf-rqm-twkyl-bwsh-blskndry01227713067tsl-ln/
> is invalid now when I browse, not logged in as an admin. However are we
> sure Google's not finding it still?
> 
> It appears that whoever is doing this is doing it all over the internet
> from the search too. There's posts on youtube, vimeo, adobe forums,
> facebook...
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/ros-infrastructure/answers.ros.org/issues/76#issuecomment-46772714
> .

## 

Askbot
Valparaiso, Chile
skype: evgeny-fadeev

**Comment by tfoote** [comment]
Looking a little close it appears that it's just google being really quick
to index the site, in the 4-8 hours it takes for us to clean up the spam.

The new block and delete button is great! I just cleaned up ~ 100 posts
from overnight in less than 5 minutes.

From the recent behavior I think flood control might be reasonable. One of
the posters posted 68 questions overnight. If we make it 1 question every 5
minutes max or something like that it would make the mass postings much
harder. Or we could allow multiple questions but require a longer backoff.
Maybe 3 per hour sort of thing while a watched user.

Other forum spam fighting techniques are listed at:
https://en.wikipedia.org/wiki/Forum_spam
​

**Comment by dornhege** [comment]
A huge +1 for "block user and delete all content".

I guess now it's moderate and see? That bot seems quite resilient.

**Comment by gavanderhoorn** [comment]
On 23-6-2014 13:40, Christian Dornhege wrote:

> A huge +1 for "block user and delete all content".

I'm guessing this is only for moderators?

**Comment by NikolausDemmel** [comment]
> > On 23-6-2014 13:40, Christian Dornhege wrote: A huge +1 for "block user and delete all content".
> > I'm guessing this is only for moderators?

I guess so, I don't see this with 1000+ karma. Spam is getting out of hand, almost every time I check the website it is mostly Spam posts. Especially when it is night time in California.

Now we also get English SPAM and from lots of different account at the same time as well.

Limiting posts to 5/day or something for watched users sounds like a good idea.

I didn't quite get if there is a catcha now for registration, or not?

**Comment by NikolausDemmel** [comment]
Also, IMHO relying on some external service like Aksimet sounds like a good idea.

**Comment by gavanderhoorn** [comment]
On 29-6-2014 12:14, Nikolaus Demmel wrote:

> > > On 23-6-2014 13:40, Christian Dornhege wrote: A huge +1 for "block user and delete all content".
> > > I'm guessing this is only for moderators?
> 
> I guess so, I don't see this with 1000+ karma. Spam is getting out of hand, almost every time I check the website it is mostly Spam posts. Especially when it is night time in California.

I must admit I'm completely ignorant of the current setup, but are there
moderators in the different time zones?

**Comment by dornhege** [comment]
This is getting out of hand. I just deleted multiple user accounts with a total of more than 1000 questions and new accounts are coming in by the minute. Now instead of deleting dozens of questions it's manually deleting dozens of accounts manually.

Rate limiting is reasonable - I never had more than 1 question/day and I can't imagine someone needing more than 3.

Still, the bot is bypassing any registration hurdles there are. Unless there is a proper solution to that right now I think the only way is moderating new user questions (at least the first one). This won't deter new users any more than a site full of spam.

**Comment by tfoote** [comment]
Yeah, I did the same and so did @mikeferguson

I also just got some alerts for an answer which was spam so they're exploring more options too.

**Comment by evgenyfadeev** [comment]
I'm working exactly on this feature now - the pre moderation.

On Sun, Jun 29, 2014 at 1:39 PM, Tully Foote notifications@github.com
wrote:

> Yeah, I did the same and so did @mikeferguson
> https://github.com/mikeferguson
> 
> I also just got some alerts for an answer which was spam so they're
> exploring more options too.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/ros-infrastructure/answers.ros.org/issues/76#issuecomment-47459772
> .

## 

Askbot
Valparaiso, Chile
skype: evgeny-fadeev

**Comment by 130s** [comment]
Meanwhile I usually `flag offensive` and delete posts (and hit the daily max (20 for me) of doing so). Is it the best that normal registered users can do?

**Comment by evgenyfadeev** [comment]
Isaac, as a normal user, this is all you can do.
Actually don't bother flagging - just delete the posts.

The moderators can also delete content of users in bulk and block users.

On Sun, Jun 29, 2014 at 9:10 PM, Isaac Isao Saito notifications@github.com
wrote:

> Meanwhile I usually flag offensive and delete posts (and hit the daily
> max (20 for me) of doing so). Is it the best that normal registered users
> can do?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/ros-infrastructure/answers.ros.org/issues/76#issuecomment-47485184
> .

## 

Askbot
Valparaiso, Chile
skype: evgeny-fadeev

**Comment by mehditlili** [comment]
There is a new attack right now from some Indian astrologers. Isn't there a feature that permits users to flag other users and not only their questions? A user flagged by 5 or more other users could be put on hold until a moderator checks what is going on.

**Comment by ccapriotti** [comment]
Domhege suggested I posted here; people, if you needs hands on deck to moderate those messages, I am volunteering, supposing the system allows for it.

**Comment by dlaz** [comment]
So far, it doesn't seem like anything has been able to proactively stop these spam attacks. That said, humans in the loop is probably our best bet, and there are clearly lots of community members willing to help. I wonder if imposing some automatic consequences for users with flagged posts would help. For example, a user with 3 flagged questions in the past day is rate limited, or has to fill out a captcha for any new questions. This can at least keep things manageable until an admin has a chance to make a decision.

**Comment by ccapriotti** [comment]
IMH, captcha for registration immediately, along with close monitoring of
new members is the way to go.

Now, something else is also concerning me: deleting all the crap from the
history on the web. But that is for later, of course.

Again, count on me.

An idea for monitoring: is there a way to create a special level of
mediation, to be given to a "lot of people" with the only objective of
stopping this (type of) attack ?

Infantry...  We take the bullet for you guys, and leave only the serious
stuff, pre-filtered, for the system or you guys. Task-force-like, this
could be a team that is called to action whenever there is an outburst.

Food for thought.

**Comment by NikolausDemmel** [comment]
> So far, it doesn't seem like anything has been able to proactively stop these spam attacks. That said, humans in the loop is probably our best bet

Am I wrong or is the only thing that has been tried captcha for registration? Aren't there lots of tools to try to detect spam automatically, e.g. Akismet?

If there are lack of resources to implement this right now, moderating new users should be done (temporarily) as immediate action.

**Comment by dlaz** [comment]
This is getting out of hand. The first 3.5 pages are currently spam, and I'm sure Tully can't devote all his time to dealing with it. There needs to be a short term fix until something is worked out - temporarily disable user registration, make a few more admins, etc.

**Comment by mikeferguson** [comment]
I just cleaned up the existing spam. Perhaps, for this particular spammer, a quick check of "the word marriage exists in this post" would suffice to auto block them...

**Comment by ccapriotti** [comment]
People, if this is open to a vote, I agree with suspending new
subscriptions for a few days.

All necessary apologies to users and action.

ROS' image is at stake here. Not dealing with this properly may cause
damage beyond repair.

Again, I am willing to help.

**Comment by evgenyfadeev** [comment]
Hi guys,

Please let me know if you want to implement the new users
registration freeze.

It should take me two days maximum to finish pre-moderation.

Best regards,
Evgeny.

On Tue, Jul 1, 2014 at 3:16 PM, ccapriotti notifications@github.com wrote:

> People, if this is open to a vote, I agree with suspending new
> subscriptions for a few days.
> 
> All necessary apologies to users and action.
> 
> ROS' image is at stake here. Not dealing with this properly may cause
> damage beyond repair.
> 
> Again, I am willing to help.
> On Jul 1, 2014 7:15 PM, "Michael Ferguson" notifications@github.com
> wrote:
> 
> > I just cleaned up the existing spam. Perhaps, for this particular
> > spammer,
> > a quick check of "the word marriage exists in this post" would suffice
> > to
> > auto block them...
> > 
> > —
> > Reply to this email directly or view it on GitHub
> > <
> > https://github.com/ros-infrastructure/answers.ros.org/issues/76#issuecomment-47684039>
> > 
> > .
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/ros-infrastructure/answers.ros.org/issues/76#issuecomment-47691232
> .

## 

Askbot
Valparaiso, Chile
skype: evgeny-fadeev

**Comment by mikeferguson** [comment]
I don't think turning off new registrations if a very good idea -- in the last 24 hours, 6 people have signed up and asked legitimate questions. We've been dealing with this for several weeks now, two more days won't be that much worse.

**Comment by ccapriotti** [comment]
Michael, I respect your point, but honestly, I myself considered
unsubscribing.

I came to my senses and simply switched off receiving emails from the list
while this is still going on, but you can imagine the wide range of
reactions this event is generating.

Six subscription in the last 24 hours is good, specially with valid
questions, and that number tells me that moderating newcomers is feasible,
if the load is distributed, preferably among people in different time zones
around the globe.

**Comment by mikeferguson** [comment]
@ccapriotti So the part of your response that shocks me is: emails work for you? You get notifications? I was under the assumption that nobody was getting any emails anymore, because non of my subscriptions appear to work (and for a long time, none of them did for anybody). So  yes, I could totally understand lots and lots of emails being annoying -- I was completely unaware that any subscriptions worked for anyone....

**Comment by ccapriotti** [comment]
Ah... never mind my last email.

I hadn't read Tully's post yet.

**Comment by tfoote** [comment]
I've recruited several new moderators and it seems to have prevented the buildup of spam. @evgenyfadeev I don't think we need to freeze new user registration. I really hope that if the spammers will not find our site useful if their content does not stay. And with the more moderators approving first posts hopefully can be prompt as well. 

@mikeferguson As far as i know subscriptions should be working. Please open a ticket if you're are not behaving as you expect.

**Comment by dornhege** [comment]
One quick note for new user's post moderation: It might be good to have an active trigger, e.g. an email to all moderators, so that questions will be activated as soon as possible. As that would only be for preventing spam it just applies to the first question of a new user and the traffic should be low.

**Comment by tfoote** [comment]
Even though we're being pretty prompt about deleting content. Google's apparently even faster at indexing it. This search shows 2890 results for "love" on ros.org https://www.google.com/search?q=site%3Aros.org+love

In looking into google's webmaster tools a related issue is the sitemap is invalid #84 We also talked about adding blocked urls to the robots.txt which might help too, but it appears they have a max size of 500k so we can't just add all the blocked pages. Hopefully with #84 will help.

**Comment by evgenyfadeev** [comment]
Fixed the sitemap issue.

On Wed, Jul 2, 2014 at 7:37 PM, Tully Foote notifications@github.com
wrote:

> Even though we're being pretty prompt about deleting content. Google's
> apparently even faster at indexing it. This search shows 2890 results for
> "love" on ros.org https://www.google.com/search?q=site%3Aros.org+love
> 
> In looking into google's webmaster tools a related issue is the sitemap is
> invalid #84
> https://github.com/ros-infrastructure/answers.ros.org/issues/84 We also
> talked about adding blocked urls to the robots.txt which might help too,
> but it appears they have a max size of 500k so we can't just add all the
> blocked pages. Hopefully with #84
> https://github.com/ros-infrastructure/answers.ros.org/issues/84 will
> help.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/ros-infrastructure/answers.ros.org/issues/76#issuecomment-47846344
> .

## 

Askbot
Valparaiso, Chile
skype: evgeny-fadeev

**Comment by skohlbr** [comment]
More spam coming in. I´m also available for moderation.

**Comment by BennyRe** [comment]
With my new moderator rights I just deleted 5 spam posts.
But these posts haven't been posted by new users! The user accounts were real. One of them had over 1700 karma.

Do we have a security gap that allows hijacking user accounts?

**Comment by gavanderhoorn** [comment]
On 3-7-2014 9:10, BennyRe wrote:

> With my new moderator rights I just deleted 5 spam posts.
> But these posts haven't been posted by new users! The user accounts were real. One of them had over 1700 karma.
> 
> Do we have a security gap that allows hijacking user accounts?

Are you sure that that account posted the spam, or did it just report
(flag) the posts?

I find the moderation interface sometimes a bit unintuitive, and have
had to look twice to make sure I wasn't blocking the user reporting the
posts.

**Comment by evgenyfadeev** [comment]
If you have moderator rights, please go to the offending user's profile ->
moderation
and click "block and delete content".

On Thu, Jul 3, 2014 at 4:13 AM, G.A. vd. Hoorn notifications@github.com
wrote:

> On 3-7-2014 9:10, BennyRe wrote:
> 
> > With my new moderator rights I just deleted 5 spam posts.
> > But these posts haven't been posted by new users! The user accounts were
> > real. One of them had over 1700 karma.
> > 
> > Do we have a security gap that allows hijacking user accounts?
> 
> Are you sure that that account posted the spam, or did it just report
> (flag) the posts?
> 
> I find the moderation interface sometimes a bit unintuitive, and have
> had to look twice to make sure I wasn't blocking the user reporting the
> posts.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/ros-infrastructure/answers.ros.org/issues/76#issuecomment-47874102
> .

## 

Askbot
Valparaiso, Chile
skype: evgeny-fadeev

**Comment by BennyRe** [comment]
Oh sorry.

I indeed thought the user that has reported the spam was the spammer, because of the user avatar.

**Comment by tfoote** [comment]
@BennyRe please make sure to unblock that user.

**Comment by ccapriotti** [comment]
How about raising the limit of "offensive flags" we mortals can use for a
while ?

I exhausted my 20 in a single attack from our beloved guru.

**Comment by dornhege** [comment]
There should not be a need for that. It is sufficient to flag only one question per spam account. Moderators will block this user and delete all its content in one action.

**Comment by skohlbr** [comment]
Requesting moderator rights (my Karma should be sufficient for that :) ). There´s 4 pages of spam since more than an hour.

**Comment by trainman419** [comment]
This new user was able to create 310 posts before I blocked them: http://answers.ros.org/users/19480/oteiu/ 

They had "watched" status, but I don't think the current captcha is working properly, or it's too easy to defeat.

**Comment by dornhege** [comment]
I just blocked the first spammer from the moderation queue before it appeared on the site. +1 !

I'd like to have some more visual separation (hline, vspace, block, etc.) between entries in the moderation queue, so that it is clear which user/checkbox belongs to a post. I don't want to accidentally block a real user.

**Comment by NikolausDemmel** [comment]
At the moment in my moderation queue for "flagged items" it says there is 1 item, but it does not list any items. 

Also, since recently in the "Forum responses" section the selection buttons ("all","none") as well as the checkboxes next to the items are missing such that I cannot discard items any longer.

**Comment by evgenyfadeev** [comment]
True, there is an issue of one item not being shown, I will have a look at
that.

Regarding the "forum responses" - I've decided to change how this works.
The "red alert" will be hanging for just some time after you look at the
responses page
and will automatically clear. There will be nothing to delete, to make it
simpler.

Best regards,
Evgeny.

On Wed, Jul 9, 2014 at 11:29 AM, Nikolaus Demmel notifications@github.com
wrote:

> At the moment in my moderation queue for "flagged items" it says there is
> 1 item, but it does not list any items.
> 
> Also, since recently in the "Forum responses" section the selection
> buttons ("all","none") as well as the checkboxes next to the items are
> missing such that I cannot discard items any longer.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/ros-infrastructure/answers.ros.org/issues/76#issuecomment-48478331
> .

## 

Askbot
Valparaiso, Chile
skype: evgeny-fadeev

**Comment by NikolausDemmel** [comment]
I'm not sure what you mean by "red alert"? I assume you are referring to the red envelope indicating new forum responses. AFAIR that alert used to always clear as soon as you looked at your responses. However I did like the ability to delete selective response items from the inbox to remind me where I haven't responded immediatly and would like to maybe do so later on.

Just to be clear, this is how my inbox looks: https://www.dropbox.com/s/vo306sk1fb1qr0a/Screenshot%202014-07-09%2016.40.23.png

**Comment by evgenyfadeev** [comment]
Yes, the red flag. I'll need to do some work to auto-clear this envelope
after
you visit the notifications list.

I think that functions to delete items in the notifications list makes
users think
that they must do some work there and make it appear more complicated,
so I'd like to try implementing this a bit differently.

On Wed, Jul 9, 2014 at 11:41 AM, Nikolaus Demmel notifications@github.com
wrote:

> I'm not sure what you mean by "red alert"? I assume you are referring to
> the red envelope indicating new forum responses. AFAIR that alert used to
> always clear as soon as you looked at your responses. However I did like
> the ability to delete selective response items from the inbox to remind me
> where I haven't responded immediatly and would like to maybe do so later on.
> 
> Just to be clear, this is how my inbox looks:
> https://www.dropbox.com/s/vo306sk1fb1qr0a/Screenshot%202014-07-09%2016.40.23.png
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/ros-infrastructure/answers.ros.org/issues/76#issuecomment-48480044
> .

## 

Askbot
Valparaiso, Chile
skype: evgeny-fadeev

**Comment by evgenyfadeev** [comment]
The layout will also be cleaned up.

On Wed, Jul 9, 2014 at 11:44 AM, Evgeny Fadeev evgeny.fadeev@gmail.com
wrote:

> Yes, the red flag. I'll need to do some work to auto-clear this envelope
> after
> you visit the notifications list.
> 
> I think that functions to delete items in the notifications list makes
> users think
> that they must do some work there and make it appear more complicated,
> so I'd like to try implementing this a bit differently.
> 
> On Wed, Jul 9, 2014 at 11:41 AM, Nikolaus Demmel <notifications@github.com
> 
> > wrote:
> > 
> > I'm not sure what you mean by "red alert"? I assume you are referring to
> > the red envelope indicating new forum responses. AFAIR that alert used to
> > always clear as soon as you looked at your responses. However I did like
> > the ability to delete selective response items from the inbox to remind me
> > where I haven't responded immediatly and would like to maybe do so later on.
> > 
> > Just to be clear, this is how my inbox looks:
> > https://www.dropbox.com/s/vo306sk1fb1qr0a/Screenshot%202014-07-09%2016.40.23.png
> > 
> > —
> > Reply to this email directly or view it on GitHub
> > https://github.com/ros-infrastructure/answers.ros.org/issues/76#issuecomment-48480044
> > .
> 
> ## 
> 
> Askbot
> Valparaiso, Chile
> skype: evgeny-fadeev

## 

Askbot
Valparaiso, Chile
skype: evgeny-fadeev

**Comment by NikolausDemmel** [comment]
I don't think notification/mail inboxes are that hard to understand for users in general. But I will let you do your thing :-)

Thanks!

**Comment by dornhege** [comment]
Right now it seems that when I "block user and delete all posts" it will only remove that one post from the moderation queue. The rest of the spam questions stay unless manually removed. I'd suggest removing already deleted questions from the moderation queue.

**Comment by evgenyfadeev** [comment]
It should also clear other posts by the same user, but not touch posts by
other users.

On Wed, Jul 9, 2014 at 12:01 PM, Christian Dornhege <
notifications@github.com> wrote:

> Right now it seems that when I "block user and delete all posts" it will
> only remove that one post from the moderation queue. The rest of the spam
> questions stay unless manually removed. I'd suggest removing already
> deleted questions from the moderation queue.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/ros-infrastructure/answers.ros.org/issues/76#issuecomment-48484564
> .

## 

Askbot
Valparaiso, Chile
skype: evgeny-fadeev

**Comment by NikolausDemmel** [comment]
The posts get deleted, but it seems they don't disappear from the moderation queue.

**Comment by evgenyfadeev** [comment]
Yes, I see some of that, thanks.

On Wed, Jul 9, 2014 at 8:11 PM, Nikolaus Demmel notifications@github.com
wrote:

> The posts get deleted, but it seems they don't disappear from the
> moderation queue.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/ros-infrastructure/answers.ros.org/issues/76#issuecomment-48547008
> .

## 

Askbot
Valparaiso, Chile
skype: evgeny-fadeev

**Comment by NikolausDemmel** [comment]
Are new users supposed to be moderated right now? I had a bunch of bunch of SPAM and a valid question in the moderation queue earlier, but now we just had new spamm users being able to post again.

**Comment by evgenyfadeev** [comment]
Yes, all posts of "watched" users are on the queue.
For those - that are not spam - even if they are perhaps crappy posts
approve those users and you won't see them on the queue any more.

That is you have to use more discretion when moderating as some
of the posts there are good.

On Thu, Jul 10, 2014 at 6:48 AM, Nikolaus Demmel notifications@github.com
wrote:

> Are new users supposed to be moderated right now? I had a bunch of bunch
> of SPAM and a valid question in the moderation queue earlier, but now we
> just had new spamm users being able to post again.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/ros-infrastructure/answers.ros.org/issues/76#issuecomment-48584975
> .

## 

Askbot
Valparaiso, Chile
skype: evgeny-fadeev

**Comment by dornhege** [comment]
There just was spam on the main page that was also in the moderation queue.

The account had 1 karma and was a "registered user".

**Comment by dornhege** [comment]
Just to be save: What will happen if I check the box when someone "marked offensive" a post and I "delete posts and block users" - will this block the user that posted the spam or the user that flagged it. Currently it shows the name and icon of the flagging user.

**Comment by evgenyfadeev** [comment]
In that case post is deleted of the user who made it, not the one who
flagged.

I will fix the presentation of the samples.

On Thu, Jul 10, 2014 at 7:30 AM, Christian Dornhege <
notifications@github.com> wrote:

> Just to be save: What will happen if I check the box when someone "marked
> offensive" a post and I "delete posts and block users" - will this block
> the user that posted the spam or the user that flagged it. Currently it
> shows the name and icon of the flagging user.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/ros-infrastructure/answers.ros.org/issues/76#issuecomment-48588659
> .

## 

Askbot
Valparaiso, Chile
skype: evgeny-fadeev

**Comment by evgenyfadeev** [comment]
It is not showing whether the user is "approved" or "watched", I will fix
this.

On Thu, Jul 10, 2014 at 7:27 AM, Christian Dornhege <
notifications@github.com> wrote:

> There just was spam on the main page that was also in the moderation queue.
> 
> The account had 1 karma and was a "registered user".
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/ros-infrastructure/answers.ros.org/issues/76#issuecomment-48588398
> .

## 

Askbot
Valparaiso, Chile
skype: evgeny-fadeev

**Comment by NikolausDemmel** [comment]
> Yes, all posts of "watched" users are on the queue. For those - that are not spam - even if they are perhaps crappy posts approve those users and you won't see them on the queue any more. 

This is how I expected things to work.

However spam of (some of the) unapproved users still gets through to the main page.

**Comment by evgenyfadeev** [comment]
This is because we are in the "audit" mode - showing all posts, but
moderating them
after the fact.

Some more work is needed to implement "premoderation".

Later we'll add statistical classifier of spam.

On Thu, Jul 10, 2014 at 7:37 AM, Nikolaus Demmel notifications@github.com
wrote:

> Yes, all posts of "watched" users are on the queue. For those - that are
> not spam - even if they are perhaps crappy posts approve those users and
> you won't see them on the queue any more.
> 
> This is how I expected things to work.
> 
> However spam of (some of the) unapproved users still gets through to the
> main page.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/ros-infrastructure/answers.ros.org/issues/76#issuecomment-48589170
> .

## 

Askbot
Valparaiso, Chile
skype: evgeny-fadeev

**Comment by NikolausDemmel** [comment]
Ah ok, I though "premoderation" was supposed to be in effect already. Fine.

(I'm not sure I understand the point of showing all new users' posts in the moderation queue if they go to main page anyway. I guess it is as easy/fast/slow for moderators to pick out spam from the main page than it is to pick out real questions from the moderation queue, but probably I'm just not getting it. If this is just an intermediary step towards "premoderation" and other techniques anyway, then no further explaining/discussion needed.)

**Comment by evgenyfadeev** [comment]
The "premoderation" (not enabled on your site yet) kind of works
 but has an issue that the new posts appear to
be "lost" - they are not shown to even the author.

I need to add display of moderated posts only to authors, then we can enable
premoderation.

It is not intended to show bad posts on the main page, but is a feature
of the method used.

On Thu, Jul 10, 2014 at 7:47 AM, Nikolaus Demmel notifications@github.com
wrote:

> Ah ok, I though "premoderation" was supposed to be in effect already. Fine.
> 
> (I'm not sure I understand the point of showing all new users' posts in
> the moderation queue if they go to main page anyway. I guess it is as
> easy/fast/slow for moderators to pick out spam from the main page than it
> is to pick out real questions from the moderation queue, but probably I'm
> just not getting it. If this is just an intermediary step towards
> "premoderation" and other techniques anyway, then no further
> explaining/discussion needed.)
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/ros-infrastructure/answers.ros.org/issues/76#issuecomment-48589914
> .

## 

Askbot
Valparaiso, Chile
skype: evgeny-fadeev

**Comment by NikolausDemmel** [comment]
Some more info about the issue raised above, where "block user and delete content" does not clear the entries from the moderation queue, neccessitating repeated handling of the same posts:

Assume that a bunch of posts from a new spam user are in the moderation queue and I select the first post and hit "block users and delete all content", the user and all its posts always get blocked/deleted and all posts usually disappear from the moderation queue. This is just as expected. However, sometimes a few posts remain in the queue and only get removed when you select them and hit "reject for reason SPAM". I'm not sure if this is related to whether the post has been flagged by someone or not.

On the other hand, when I see a bunch of spam posts from a new user and delete them through the moderation page of the user's profile (not through the moderation queue), then these posts seem to remain in the moderation queue (they are correctly deleted and the users blocked, though).

**Comment by NikolausDemmel** [comment]
It seems that now a lot of posts appear in the moderation queue that shouldn't. E.g. right now there are two posts from "registered users" in the moderation queue. One has 2 posts, one 6.

**Comment by tfoote** [comment]
We've resolved the major issue by premoderation. I've created some followups for improvements. Closing this ticket.

**Comment by gavanderhoorn** [comment]
Nothing serious, just an update: seems the spam 'bots' have something new (and I almost fell for it):

[john read](http://answers.ros.org/users/25820/john-read/) wrote:

> Gary Servin has provided you quite relevant link which contains many informative points regarding your question. `ipad applications developer`

It's a comment on an actual question ([this one](http://answers.ros.org/question/214736/arduino-sensor-data-recombined-and-published-on-master-computer/)), referencing an actual user (that also commented on the same question). Notice the spam link at the end of the (boilerplate) comment.
